ekncontent/ekncontent/eknc-content-object-model.c

```diff
@@ -0,0 +1,578 @@
+      break;
+
+    case PROP_TITLE:
+      g_free (priv->title);
+      priv->title = g_value_dup_string (value);
+      break;
+
+    case PROP_ORIGINAL_TITLE:
+      g_free (priv->original_title);
+      priv->original_title = g_value_dup_string (value);
+      break;
+
+    case PROP_ORIGINAL_URI:
+      g_free (priv->original_uri);
+      priv->original_uri = g_value_dup_string (value);
+      break;
+
+    case PROP_THUMBNAIL_URI:
+      g_free (priv->thumbnail_uri);
+      priv->thumbnail_uri = g_value_dup_string (value);
+      break;
+
+    case PROP_LANGUAGE:
+      g_free (priv->language);
+      priv->language = g_value_dup_string (value);
+      break;
+
+    case PROP_COPYRIGHT_HOLDER:
+      g_free (priv->copyright_holder);
+      priv->copyright_holder = g_value_dup_string (value);
+      break;
+
+    case PROP_SOURCE_URI:
+      g_free (priv->source_uri);
+      priv->source_uri = g_value_dup_string (value);
+      break;
+
+    case PROP_CONTENT_TYPE:
+      g_free (priv->content_type);
+      priv->content_type = g_value_dup_string (value);
+      break;
+
+    case PROP_SYNOPSIS:
+      g_free (priv->synopsis);
+      priv->synopsis = g_value_dup_string (value);
+      break;
+
+    case PROP_LAST_MODIFIED_DATE:
+      g_free (priv->last_modified_date);
+      priv->last_modified_date = g_value_dup_string (value);
+      break;
+
+    case PROP_LICENSE:
+      g_free (priv->license);
+      priv->license = g_value_dup_string (value);
+      break;
+
+    case PROP_FEATURED:
+      priv->featured = g_value_get_boolean (value);
+      break;
+
+    case PROP_TAGS:
+      priv->tags = g_value_dup_variant (value);
+      break;
+
+    case PROP_RESOURCES:
+      priv->resources = g_value_dup_variant (value);
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+eknc_content_object_model_constructed (GObject *object)
+{
+  EkncContentObjectModel *self = EKNC_CONTENT_OBJECT_MODEL (object);
+  EkncContentObjectModelPrivate *priv = eknc_content_object_model_get_instance_private (self);
+
+  if (g_strcmp0 (priv->ekn_id, "") == 0)
+    {
+      // Note: This is only for ensuring the invariant of "each model has an
+      // EKN ID" in tests. It is illegal to create a model in production code
+      // with no EKN ID.
+      g_autofree gchar *sum = NULL;
+      static guint64 MOCKED_OBJECTS = 0;
+      sum = g_compute_checksum_for_data (G_CHECKSUM_SHA1,
+                                         (guchar *) &MOCKED_OBJECTS,
+                                         sizeof (MOCKED_OBJECTS));
+      MOCKED_OBJECTS++;
+      priv->ekn_id = g_strconcat ("ekn:///", sum, NULL);
+    }
+}
+
+static void
+eknc_content_object_model_finalize (GObject *object)
+{
+  EkncContentObjectModel *self = EKNC_CONTENT_OBJECT_MODEL (object);
+  EkncContentObjectModelPrivate *priv = eknc_content_object_model_get_instance_private (self);
+
+  g_free (priv->ekn_id);
```

Comment:
Ohh I see, yeah that make sense, probably good for dangling refs to the object. I'm sold. And gnome-builder seems to use this style everywhere.